### PR TITLE
feat: audit

### LINKED ISSUE #1153: Support audit
There's a new command available called "audit" since NPM npm@5.10.0 and npm@6. It requires the presence of both package.json and package-lock.json meaning it cannot be just a pass through. It would be great if PNPM supported it as well.

> This version of npm adds a new command, npm audit, which will run a security audit of your project's dependency tree and notify you about any actions you may need to take.

> As part of this change, the npm CLI now sends scrubbed and cryptographically anonymized metadata about your dependency tree to your configured registry, to allow notifying you about the existence of critical security flaws.

https://blog.npmjs.org/post/173719309445/npm-audit-identify-and-fix-insecure

### PR DESCRIPTION
**Pull request by zkochan**
close #1153

### CODE REVIEW

**Comment by octogonz** [comment]
Awesome!  👏

**Comment by zkochan** [comment]
npm has an `--audit-level <severity>` flag. Yarn has a `--level <severity>` flag.

I guess I'll go with the npm version as it is easier to implement when the config setting and the CLI flag have the same name (with the current setup of pnpm).

File: packages/pnpm/src/cmd/help.ts [hunk 1]
```diff
@@ -84,6 +84,24 @@ const FILTERING = {
...
+        descriptionLists: [
+          {
+            title: 'Options',
+
+            list: [
+              {
+                description: 'Output audit report in JSON format',
+                name: '--json',
+              },
+            ],
```
Comment (ExE-Boss) [inline packages/pnpm/src/cmd/help.ts]: This should probably mention `‑‑audit‑level` as&nbsp;well.

File: packages/audit/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
+import { Lockfile } from '@pnpm/lockfile-types'
+import { nameVerFromPkgSnapshot } from '@pnpm/lockfile-utils'
+import { lockfileWalkerGroupImporterSteps, LockfileWalkerStep } from '@pnpm/lockfile-walker'
+import fetch from 'node-fetch'
```
Comment (ExE-Boss) [inline packages/audit/src/index.ts]: Why&nbsp;not&nbsp;use `@pnpm/fetch`&nbsp;here, like&nbsp;what&nbsp;every&nbsp;other&nbsp;package in&nbsp;this&nbsp;repository&nbsp;does?

Running `pnpm -r list node-fetch --prod` reveals&nbsp;that only&nbsp;`@pnpm/fetch` depends&nbsp;on&nbsp;it, whereas&nbsp;`pnpm -r list @pnpm/fetch --prod`&nbsp;reveals&nbsp;that it&nbsp;has&nbsp;3&nbsp;dependents in&nbsp;this&nbsp;repository.

**Comment by shellscape** [comment]
That was so fast!!

File: packages/audit/package.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  "author": "Zoltan Kochan <z@kochan.io> (https://www.kochan.io/)",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/pnpm/pnpm/issues"
+  },
+  "homepage": "https://pnpm.js.org",
+  "devDependencies": {
+    "@pnpm/audit": "link:",
+    "@pnpm/lockfile-file": "workspace:3.0.1",
+    "@types/node-fetch": "2.5.3",
```
Comment (ExE-Boss) [inline packages/audit/package.json:33]: I&nbsp;believe&nbsp;that this&nbsp;is no&nbsp;longer&nbsp;necessary.

Comment (zkochan) [reply packages/audit/package.json:33]: yes, thanks

**Pull request merged**